### PR TITLE
Add uninstall functionality for plugins

### PR DESCRIPTION
**Pull request by karthik2804**
Signed-off-by: karthik Ganeshram <karthik.ganeshram@fermyon.com>

### CODE REVIEW

File: crates/plugins/src/uninstall.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+use crate::{get_manifest_file_name, PLUGIN_MANIFESTS_DIRECTORY_NAME};
+use anyhow::Result;
+use std::{fs, path::PathBuf};
+pub struct PluginUninstaller {
+    plugin_name: String,
+    plugins_dir: PathBuf,
+}
+
+impl PluginUninstaller {
+    pub fn new(plugin_name: &String, plugins_dir: PathBuf) -> Self {
```
Comment (kate-goldenring) [inline crates/plugins/src/uninstall.rs]: Why `&String`? This should probably be `&str`. Does `cargo clippy` catch this by any chance?

**Review by kate-goldenring — APPROVED** [review]
just one nit

File: crates/plugins/src/uninstall.rs (hunk 1 above)
Comment (karthik2804) [reply crates/plugins/src/uninstall.rs]: Good catch. `cargo clippy` does not throw any error but I will make the change.

File: crates/plugins/src/git.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ impl GitSource {
         let mut git = Command::new("git");
         git.args([
             "clone",
-            &self.source_url.to_string(),
+            self.source_url.as_ref(),
```
Comment (kate-goldenring) [inline crates/plugins/src/git.rs:31]: Can this be &self.source_url instead of `as_ref`?

Comment (karthik2804) [reply crates/plugins/src/git.rs:31]: It is a URL that needs to be converted to a string and Clippy complains when I write it as  `&self.source_url.to_string()`  to replace it as `self.source_url.as_ref()`

Comment (kate-goldenring) [reply crates/plugins/src/git.rs:31]: makes sense! thanks!

**Pull request merged**